tests/Aspire.Workload.Tests/WorkloadTestsBase.cs

```diff
@@ -0,0 +1,263 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Globalization;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Text.RegularExpressions;
+using Microsoft.Playwright;
+using Xunit;
+using Xunit.Abstractions;
+using static Aspire.Workload.Tests.TestExtensions;
+
+namespace Aspire.Workload.Tests;
+
+public class WorkloadTestsBase
+{
+    private static Lazy<IBrowser> Browser => new(CreateBrowser);
+    protected readonly TestOutputWrapper _testOutput;
+
+    public WorkloadTestsBase(ITestOutputHelper testOutput)
+        => _testOutput = new TestOutputWrapper(testOutput);
+
+    private static IBrowser CreateBrowser()
+    {
+        var t = Task.Run(async () =>
+        {
+            var playwright = await Playwright.CreateAsync();
+            string? browserPath = EnvironmentVariables.BrowserPath;
+            if (!string.IsNullOrEmpty(browserPath) && !File.Exists(browserPath))
+            {
+                throw new FileNotFoundException($"Browser path BROWSER_PATH='{browserPath}' does not exist");
+            }
+
+            BrowserTypeLaunchOptions options = new()
+            {
+                Headless = false,
+                ExecutablePath = browserPath
+            };
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX) && string.IsNullOrEmpty(browserPath))
+            {
+                var probePath = "/Applications/Google Chrome.app/Contents/MacOS/Google Chrome";
+                if (File.Exists(probePath))
+                {
+                    options.ExecutablePath = probePath;
+                }
+            }
+            return await playwright.Chromium.LaunchAsync(options).ConfigureAwait(false);
+        });
+
+        // default timeout for LaunchAsync is 30secs, so pick something larger than that
```

Comment:
comment seems off from the code

Reply:
Updated the comment. Please lemme know if it still seems off.